lib/hubspot/utils.rb

```diff
@@ -107,12 +107,8 @@ def compare_property_lists(klass, source, target)
       end
 
       def with_hapikey(hapikey)
-        begin
-          Hubspot.configure(hapikey: hapikey) unless hapikey.blank?
-          yield if block_given?
-        ensure
-          Hubspot.configure(hapikey: ENV['HUBSPOT_API_KEY']) unless hapikey.blank?
-        end
+        Hubspot.configure(hapikey: hapikey)
+        yield if block_given?
```

Comment:
Updating the config to raise when the an authentication approach is not set resulted in a few tests failing because the ensure block is setting the `hapikey` to a nil value.

This method currently:
sets the `hapikey` to the provided `hapikey` unless the argument is `blank`.
runs the block if a block is given
sets the `hapikey` to the env variable `HUBSPOT_API_KEY` unless the `hapikey` is `blank`.
 
I think it's safe to simplify this method to rely on an `hapikey` value being provided and not attempting to set the `hapikey` to an environment variable. This method is used by the deprecated methods `dump_properties` and `dump_properties`.